independent-projects/resteasy-reactive/client/runtime/src/main/java/org/jboss/resteasy/reactive/client/handlers/ClientCaptureCurrentContextRestHandler.java

```diff
@@ -35,21 +37,18 @@ public void handle(RestClientRequestContext requestContext) throws Exception {
     }
 
     private void captureCallerStackTrace(ClientRequestContextImpl clientRequestContext) {
-        StackTraceElement[] stackTrace = Thread.currentThread().getStackTrace();
-        List<StackTraceElement> effectiveStackTrace = new ArrayList<>(stackTrace.length);
-        boolean foundUserMethod = false;
-        // skip first trace which is Thread.getStackTrace
-        for (int i = 1; i < stackTrace.length; i++) {
-            StackTraceElement trace = stackTrace[i];
-            if (foundUserMethod) {
-                effectiveStackTrace.add(trace);
-            } else if (!trace.getClassName().startsWith(RESTEASY_REACTIVE_PACKAGE)
-                    && !trace.getClassName().contains(AUTOGENERATED_TAG)) {
-                // Skip the latest traces that starts with the "org.jboss.resteasy.reactive" package,
-                effectiveStackTrace.add(trace);
-                foundUserMethod = true;
+        List<StackTraceElement> effectiveStackTrace = StackWalker.getInstance().walk(new Function<>() {
+            @Override
+            public List<StackTraceElement> apply(Stream<StackWalker.StackFrame> stream) {
+                return stream.dropWhile(new Predicate<>() {
+                    @Override
+                    public boolean test(StackWalker.StackFrame stackFrame) {
+                        String className = stackFrame.getClassName();
+                        return className.startsWith(RESTEASY_REACTIVE_PACKAGE) || className.contains(AUTOGENERATED_TAG);
+                    }
+                }).limit(5).map(StackWalker.StackFrame::toStackTraceElement).toList();
```

Comment:
Is 5 lines enough?
this probably needs to be documented or add a config to keep the switch to the old behaviour.

Reply:
It's fine for identifying which REST Client is causing the problem and where it was called from.

It's not a complete stacktrace, but the complete one is mostly useless except for the top frames